x-pack/plugins/triggers_actions_ui/public/application/lib/action_variables.ts

```diff
@@ -115,6 +116,14 @@ function getAlwaysProvidedActionVariables(): ActionVariable[] {
     }),
   });
 
+  result.push({
+    name: AlertProvidedActionVariables.viewInAppUrl,
+    description: i18n.translate('xpack.triggersActionsUI.actionVariables.viewInAppUrlLabel', {
+      defaultMessage:
+        'The URL to the associated application that generated the alert. This will be an empty string if the server.publicBaseUrl is not configured.',
+    }),
+  });
+
```

Comment:
@elastic/response-ops-execution looking for some of your thoughts.. this PR adds a mustache variable (`{{rule.viewInAppUrl}}`) to provide a view in app url. I realize the actions form is mainly agnostic of the rule and rule type. Given this variable only renders for rule types that define a `getViewInAppRelativeUrl` function, its empty for other rule types..

I’ve been weighing options on this oddity, looking for thoughts..
1. Whenever a rule type hasn’t defined `getViewInAppRelativeUrl` function, we still provide the mustache variable but make the URL point to the rule details page (same as `rule.url`)
2. Hide the variable from the list of variables whenever it would render empty.. requires some refactoring, we probably don’t want rule action variables to be part of the actions framework and would also facilitate these dynamic show/hide ones..
3. Other options

🤔 option 1 is easy while option 2 will require some overhead / rework, but.. is probably the better option.

Reply:
Are there any rule types that really SHOULDN'T have a `viewInAppUrl`?  Index threshold / es query that are using indexes instead of data patterns don't really have an "app" per se, and perhaps not maps, but ... seems like everything else should?

So I'm tempted to go with option 1, where we say the "some rule types send you back to the rule details page if there is no specific app page available".  And then get on the rule type authors to add the string property to their rules :-)

Another option would be to treat it like the `url` one - if the value isn't otherwise defined, it would print as an empty string in mustache, and is a valid property to use in a conditional template.  So you could build conditional things like this - but it obviously seems a bit messy (the second line uses an "inverse" section, so if it's NOT set):

```
{{#rule.viewInAppUrl}}View in the Kibana app: {{.}}{{/rule.viewInAppUrl}}
{{^rule.viewInAppUrl}}{{#rule.url}}View in the Kibana Rules app: {{.}}{{/rule.url}}{{/rule.viewInAppUrl}}
```

Reply:
There are some rule types that generate their own context rule link inside the executors. Wondering if the ones that do this also generate a viewInApp url and whether they are the same URL? If so, then it seems reasonable to me to fall back to a viewInApp value that links back to the stack rule page since that's what the rule.url value defaults to (so Option 1).

Reply:
Thanks for the feedback 🙏 Seems like option 1 is best.

>Are there any rule types that really SHOULDN'T have a viewInAppUrl? Index threshold / es query that are using indexes instead of data patterns don't really have an "app" per se, and perhaps not maps, but ... seems like everything else should?

Most rule types should, or maybe in the future could. For example ES Query could someday point to discover.

>There are some rule types that generate their own context rule link inside the executors. Wondering if the ones that do this also generate a viewInApp url and whether they are the same URL? If so, then it seems reasonable to me to fall back to a viewInApp value that links back to the stack rule page since that's what the rule.url value defaults to (so Option 1).

That's my thinking, a consistent way to have a URL generated. In alert summaries, we don't have `state` or `context` so rule types can't generate URLs unless it's persisted inside the alerts-as-data document. This option allows URLs to be accessible in summaries without the AAD workaround.

Reply:
> For example ES Query could someday point to discover

That rule type already generates a `{{context.link}}` that points into Discover \o/, though I'd guess today only if you use a "data view" for the index selection, vs picking an index specifically.  Though I would expect that to be supported at some point as well, when Discover can run over plain old indices and not just Data Views.

Here's the link it just generated for me:

`https://<<hostname>>:9243/app/discover#/viewAlert/f8426620-9e65-11ed-b5c4-27d1f241d08e?from=2023-01-27T12:16:08.843Z&to=2023-01-27T17:16:08.846Z&checksum=49f4764ff7541bfe271650dfe46ef862d41f1896150995ae3771d6ef2ae84773`

I believe the checksum arg is used to check something about the dataview, to make sure it's still ok to reference it, or something.  Maybe that's needed as the links in actions, not necessarily needed (or wanted, or easily calculable?) from a "View in App" button on the Rule page?